lib/core/src/Cardano/Wallet/Primitive/Types.hs

```diff
@@ -517,6 +523,49 @@ instance FromText PoolId where
         where
             textDecodingError = Left . TextDecodingError . show
 
+-- | A stake pool owner, which is a public key encoded in bech32 with prefix
+-- ed25519_pk.
+newtype PoolOwner = PoolOwner { getPoolOwner :: ByteString }
+    deriving (Generic, Eq, Show, Ord)
+
+poolOwnerPrefix :: Bech32.HumanReadablePart
+poolOwnerPrefix = Bech32.unsafeHumanReadablePartFromText "ed25519_pk"
+
+instance NFData PoolOwner
+
+instance Buildable PoolOwner where
+    build poolId = build (toText poolId)
+
+instance ToText PoolOwner where
+    toText = Bech32.encodeLenient poolOwnerPrefix
+        . Bech32.dataPartFromBytes
+        . getPoolOwner
+
+instance FromText PoolOwner where
+    fromText t = case fmap Bech32.dataPartToBytes <$> Bech32.decode t of
+        Left err ->
+            Left $ TextDecodingError $
+            "Stake pool owner is not a valid bech32 string: "
+            <> show err
+        Right (hrp, Just bytes)
+            | hrp == poolOwnerPrefix ->
+                Right $ PoolOwner bytes
+            | otherwise ->
+                Left $ TextDecodingError $
+                "Stake pool owner has wrong prefix:"
+                <> " expected "
+                <> T.unpack (Bech32.humanReadablePartToText poolOwnerPrefix)
+                <> " but got "
+                <> show hrp
+        Right (_, Nothing) ->
+                Left $ TextDecodingError "Stake pool owner is invalid"
+
+instance FromJSON PoolOwner where
+    parseJSON = parseJSON >=> either (fail . show . ShowFmt) pure . fromText
+
+instance ToJSON PoolOwner where
+    toJSON = toJSON . toText
```

Comment:
**[question]**

It now seems a bit inconsistent that we enforce some quite strong validations on the `PoolOwner` format, whereas we still do not validate other fields (like addresses) for instance. So, what is the rationale here :thinking: ? Shall we also enforce stricter validation for addresses there?

Reply:
The `PoolOwner` is also going to be extracted from the registration certificate. It's no longer just a field of `StakePoolMetadata`, so that's why.

Reply:
Got it :+1: